manifests/server.pp

```diff
@@ -1,32 +1,36 @@
 class rsyslog::server (
-  Optional[Hash]  $global_config   = {},
-  Optional[Hash]  $legacy_config   = {},
-  Optional[Hash]  $templates       = {},
-  Optional[Hash]  $actions         = {},
-  Optional[Hash]  $inputs          = {},
-  Optional[Hash]  $custom_config   = {},
-  Optional[Hash]  $main_queue_opts = {},
-  Optional[Hash]  $modules         = {},
-  Optional[Hash]  $lookup_tables   = {},
-  Optional[Hash]  $parsers         = {},
-  Optional[Hash]  $rulesets        = {},
+  Optional[Hash] $global_config      = {},
+  Optional[Hash] $legacy_config      = {},
+  Optional[Hash] $templates          = {},
+  Optional[Hash] $actions            = {},
+  Optional[Hash] $inputs             = {},
+  Optional[Hash] $custom_config      = {},
+  Optional[Hash] $main_queue_opts    = {},
+  Optional[Hash] $modules            = {},
+  Optional[Hash] $lookup_tables      = {},
+  Optional[Hash] $parsers            = {},
+  Optional[Hash] $rulesets           = {},
+  Optional[Hash] $property_filters   = {},
+  Optional[Hash] $expression_filters = {},
```

Comment:
There is no need for Optional if we provide a default value. An empty hash is a valid datatype.

Reply:
For now I was just trying to maintain consistency.

Reply:
I agree with @bastelfreak that these parameters should not be ‘Optional’. ‘Optional’ and not defaulting to ‘undef’ is *almost* always wrong.
I also see @dhollinger’s point. Adding a couple of parameters shouldn’t require refactoring everything and consistency is important.
Maybe the refactoring should be in a separate PR? Not sure if it should come before or immediately after this one.

Reply:
I was only talking about the new params, not the existing ones.